src/audio_worklet.js

```diff
@@ -32,51 +32,119 @@ function createWasmAudioWorkletProcessor(audioParams) {
+     * multi-MB stack is passed), allocated at the *top* of the worklet's stack
+     * (and whose addresses are fixed). The 'minimum alloc' firstly stops
+     * STACK_OVERFLOW_CHECK failing (since the stack will be full, and 16 bytes
+     * being the minimum allocation size due to alignments) and leaves room for
+     * a single AudioSampleFrame as a minumum.
+     */
+    createOutputViews() {
+      // These are still alloc'd to take advantage of the overflow checks, etc.
+      var oldStackPtr = stackSave();
+      var viewDataIdx = {{{ getHeapOffset('stackAlloc(this.maxBuffers * this.bytesPerChannel)', 'float') }}};
+#if WEBAUDIO_DEBUG
+      console.log(`AudioWorklet creating ${this.maxBuffers} buffer one-time views (for a stack size of ${wwParams.stackSize} at address ${ptrToString(viewDataIdx * 4)})`);
+#endif
+      this.outputViews.length = 0;
+      for (var n = this.maxBuffers; n > 0; n--) {
+        // Added in reverse so the lowest indices are closest to the stack top
+        this.outputViews.unshift(
+          HEAPF32.subarray(viewDataIdx, viewDataIdx += this.samplesPerChannel)
+        );
+      }
+      stackRestore(oldStackPtr);
     }
 
     static get parameterDescriptors() {
       return audioParams;
     }
 
     /**
+     * Marshals all inputs and parameters to the Wasm memory on the thread's
+     * stack, then performs the wasm audio worklet call, and finally marshals
+     * audio output data back.
+     *
      * @param {Object} parameters
      */
     process(inputList, outputList, parameters) {
-      // Marshal all inputs and parameters to the Wasm memory on the thread stack,
-      // then perform the wasm audio worklet call,
-      // and finally marshal audio output data back.
+#if ALLOW_MEMORY_GROWTH
+      // recreate the output views if the heap has changed
+      if (HEAPF32.buffer != this.outputViews[0].buffer) {
+        this.createOutputViews();
+      }
+#endif
 
       var numInputs = inputList.length;
       var numOutputs = outputList.length;
 
       var entry; // reused list entry or index
       var subentry; // reused channel or other array in each list entry or index
 
-      // Calculate how much stack space is needed.
-      var bytesPerChannel = this.samplesPerChannel * {{{ getNativeTypeSize('float') }}};
-      var stackMemoryNeeded = (numInputs + numOutputs) * {{{ C_STRUCTS.AudioSampleFrame.__size__ }}};
+      // Calculate the required stack and output buffer views (stack is further
+      // split into aligned structs and the raw float data).
+      var stackMemoryStruct = (numInputs + numOutputs) * {{{ C_STRUCTS.AudioSampleFrame.__size__ }}};
+      var stackMemoryData = 0;
+      for (entry of inputList) {
+        stackMemoryData += entry.length * this.bytesPerChannel;
+      }
+      // Collect the total number of output channels (mapped to array views)
+      var outputViewsNeeded = 0;
+      for (entry of outputList) {
+        outputViewsNeeded += entry.length;
+      }
```

Comment:
The above loop multiplies by `* this.bytesPerChannel;` outside the loop, though the loop before that multiplies inside the loop. A small micro-opt would be to merge to multiply once outside. (though LMK either way)